### PR TITLE
Support a separate privacy policy on checkout.

### LINKED ISSUE #6484: Add support for a Privacy Policy checkbox
### Explain your enhancement (please be detailed)
We should add support for a separate Privacy Policy checkbox. This checkbox should be allowed to be independent of the Terms of Agreement checkbox.

It should also allow the support of a single checkbox "Terms of Agreement" but also display the content or a Privacy Policy on checkout.

It should support pulling the content of any created Privacy Policy page, which is being added to WordPress core, so that a store owner does not have to keep to copies active. It can simply pull the `page_for_privacy_policy` option, and use `get_post_field( 'post_content'...` to get the information for it.

### Justification or use case
To allow compliance for GDPR

### PR DESCRIPTION
**Pull request by cklosowski**
Fixes #6484 

Proposed Changes:
1. Add settings for privacy policy agreement, privacy policy agreement text, privacy policy page, and show privacy policy on checkout
2. Includes logic to handle showing 1 checkbox but two agreements/policies, or 2 individually.

### CODE REVIEW

File: includes/admin/settings/register-settings.php [hunk 1]
```diff
@@ -872,6 +880,34 @@ function edd_get_registered_settings() {
...
+					'show_privacy_policy_on_checkout' => array(
+						'id'   => 'show_to_privacy_policy_on_checkout',
+						'name' => __( 'Show the privacy policy on checkout', 'easy-digital-downloads' ),
+						'desc' => __( 'Display your privacy policy on checkout.', 'easy-digital-downloads' ),
+						'type' => 'checkbox',
+					),
+					'agree_privacy_page' => array(
+						'id'   => 'privacy_agree_page',
+						'name' => __( 'Privacy Agreement Page', 'easy-digital-downloads' ),
+						'desc' => __( 'If Agree to Privacy Policy is checked, enter page for the Privacy Agreement Here.', 'easy-digital-downloads' ),
```
Comment (pippinsplugins) [inline includes/admin/settings/register-settings.php]: `enter` should be `select` here.

File: includes/checkout/template.php [hunk 2]
```diff
@@ -770,21 +770,45 @@ function edd_terms_agreement() {
...
+				$agree_page      = edd_get_option( 'privacy_agree_page', get_option( 'page_for_privacy_policy' ) );
+				$agree_label     = edd_get_option( 'privacy_agree_label', __( 'Agree to Terms?', 'easy-digital-downloads' ) );
+				$agreement_text  = get_post_field( 'post_content', $agree_page );
+				?>
+
+				<?php if ( ! empty( $agreement_text ) ) : ?>
+					<div id="edd-privacy-policy" class="edd-terms" style="display:none;">
+						<?php
+						do_action( 'edd_before_privacy_policy' );
+						echo wpautop( stripslashes( $agreement_text ) );
```
Comment (pippinsplugins) [inline includes/checkout/template.php]: This should either call `do_shortcode` or should run the `the_content` filter to ensure shortcodes on the page get parsed.

Comment (cklosowski) [reply includes/checkout/template.php]: :+1:

File: includes/admin/settings/register-settings.php (hunk 1 above)
Comment (cklosowski) [reply includes/admin/settings/register-settings.php]: :+1:

File: includes/checkout/template.php (hunk 2 above)
Comment (SDavisMedia) [reply includes/checkout/template.php]: @cklosowski I'm not seeing shortcodes working here. Am I understanding correctly that they are supposed to now?

**Comment by cklosowski** [comment]
@SDavisMedia Yeah sorry I assumed it would just work. Silly me. the problem is that this is populated via AJAX, and AJAX requests have no access to the Shortcodes API. I'll come up with a work around.

**Comment by cklosowski** [comment]
@SDavisMedia all set now. Sorry I had only added `do_shortcode()` to one of the sections, not both.

**Comment by cklosowski** [comment]
@pippinsplugins we good here now?

**Pull request merged**